### PR TITLE
fix: Pass all table columns along as we need them for filtering the rows

### PR DESCRIPTION
**Pull request by juliushaertl**
Fixes https://github.com/nextcloud/tables/issues/827

Fix for an internal finding

To reproduce:
- Create a table with a single select and some other fields
- Create a view to filter for the single select but don't include it in the visible columns list
- Load the view

Would be good to cover this case with a test, but pushed to have CI running already to be able to apply a patch as a quick fix

### CODE REVIEW

**Comment by enjeck** [comment]
I'm trying to reproduce the error. What does "Create a view to filter for the single select but don't include it in the visible columns list" mean?

**Comment by juliushaertl** [comment]
Uncheck the column from the list of columns but still filter for a value (The Test column is a single select with two options):

Screenshot 2024-02-08 at 14 58 03

**Comment by enjeck** [comment]
Oh, I see

File: lib/Db/Row2Mapper.php [hunk 1]
```diff
@@ -267,12 +271,13 @@ private function getFilter(IQueryBuilder &$qb, array $filterGroup): array {
 		$filterExpressions = [];
 		foreach ($filterGroup as $filter) {
 			$columnId = $filter['columnId'];
+			$column = $this->columns[$filter['columnId']] ?? $this->allColumns[$filter['columnId']];
```
Comment (enjeck) [inline lib/Db/Row2Mapper.php:274]:
```suggestion
			$column = $this->columns[$columnId] ?? $this->allColumns[$columnId];
```

**Review by enjeck — APPROVED** [review]
Tested and works

**Pull request merged**